src/DotLiquid/Util/StrFTime.cs

```diff
@@ -16,79 +16,116 @@ public static class StrFTime
-            { "v", (dateTime, culture) => dateTime.ToString("%d-MMM-yyyy", culture).ToUpper().PadLeft(11, ' ') },
-            { "V", (dateTime, culture) => dateTime.GetIso8601WeekOfYear("V", culture) },
-            { "w", (dateTime, culture) => ((int) dateTime.DayOfWeek).ToString(culture) },
-            { "W", (dateTime, culture) => culture.Calendar.GetWeekOfYear(dateTime, CalendarWeekRule.FirstFullWeek, DayOfWeek.Monday).ToString(culture).PadLeft(2, '0') },
-            { "x", (dateTime, culture) => dateTime.ToString("MM/dd/yy", culture) },
-            { "X", (dateTime, culture) => dateTime.ToString("HH:mm:ss", culture) },
-            { "y", (dateTime, culture) => dateTime.ToString("yy", culture) },
-            { "Y", (dateTime, culture) => dateTime.ToString("yyyy", culture) },
-            { "z", (dateTime, culture) => dateTime.ToString("%K", culture).Replace(":", string.Empty) },
-            { ":z", (dateTime, culture) => dateTime.ToString("%K", culture) },
+            { "T", (dateTime, culture) => string.Format(culture, "{0:HH:mm:ss}", dateTime) },
+            { "v", (dateTime, culture) => string.Format(culture, "{0:%d-MMM-yyyy}", dateTime).ToUpper().PadLeft(11, ' ') },
+            { "x", (dateTime, culture) => string.Format(culture, "{0:MM/dd/yy}", dateTime) },
+            { "X", (dateTime, culture) => string.Format(culture, "{0:HH:mm:ss}", dateTime) },
+            { "y", (dateTime, culture) => string.Format(culture, "{0:yy}", dateTime) },
+            { "Y", (dateTime, culture) => string.Format(culture, "{0:yyyy}", dateTime) },
+            { "z", (dateTime, culture) => string.Format(culture, "{0:%K}", dateTime).Replace(":", string.Empty) },
+            { ":z", (dateTime, culture) => string.Format(culture, "{0:%K}", dateTime) },
             // ::z - not implemented
-            { "Z", (dateTime, culture) => dateTime.ToString("zzz", culture) },
+            { "Z", (dateTime, culture) => string.Format(culture, "{0:zzz}", dateTime) },
             { "%", (dateTime, culture) => "%" } // A % sign
         };
 
-        private static readonly Dictionary<string, DateTimeOffsetDelegate> OffsetFormats = new Dictionary<string, DateTimeOffsetDelegate>
+        private static readonly Dictionary<string, DateTimeDelegate> DateFormats = new Dictionary<string, DateTimeDelegate>
         {
-            { "s", (dateTimeOffset, culture) => ((long)(dateTimeOffset - new DateTimeOffset(1970, 1, 1, 0,0,0, TimeSpan.Zero)).TotalSeconds).ToString(culture) },
-            { "z", (dateTimeOffset, culture) => dateTimeOffset.ToString("%K", culture).Replace(":", string.Empty) },
-            { ":z", (dateTimeOffset, culture) => dateTimeOffset.ToString("%K", culture) },
-            { "Z", (dateTimeOffset, culture) => dateTimeOffset.ToString("zzz", culture) }
+            { "g", (dateTime, culture) => dateTime.GetIso8601WeekOfYear("g", culture) },

```

Comment:
Note to self - there is no reason why DateOnly is excluded from this functionality. TimeOnly though is another story. Existing test also is not asserting anything...